server/tests/unit/common/test_api.py

```diff
@@ -733,10 +734,38 @@ def test_tombstoned_datasets_redirect_to_data_portal(self, mock_get):
         self.assertEqual(result.status_code, 302)
         self.assertEqual(result.headers['Location'], "https://cellxgene.staging.single-cell.czi.technology.com/collections/4f098ff4-4a12-446b-a841-91ba3d8e3fa6?tombstoned_dataset_id=2fa37b10-ab4d-49c9-97a8-b4b3d80bf939") # noqa E501
 
+    @patch("server.app.app.expire_metadata_cache")
+    @patch("server.data_common.dataset_metadata.request_dataset_metadata_from_data_portal")
+    def test_metadata_cache_item_invalidated_on_errors(self, mock_dp, mock_expire):
+        mock_expire.side_effect = expire_metadata_cache
+        response_body_bad = {
+            "collection_id": "4f098ff4-4a12-446b-a841-91ba3d8e3fa6",
+            "collection_visibility": "PUBLIC",
+            "dataset_id": "2fa37b10-ab4d-49c9-97a8-b4b3d80bf939",
+            "s3_uri": f"BAD_PATH_TO_DATASET/pbmc3k.cxg",
+            "tombstoned": False,
+        }
+        mock_dp.return_value = response_body_bad
+        TEST_DATASET_URL_BASE = "/e/pbmc3k_v3.cxg"
+        url = f"{TEST_DATASET_URL_BASE}/api/v0.2/config"
+        bad_response = self.client.get(url)
+        self.assertEqual(bad_response.status_code, 404)
+        self.assertEqual(mock_expire.call_count, 1)
+        response_body_good = {
```

Comment:
is this happy path test, below, not already being tested elsewhere?

Reply:
nit: add a newline above this to separate the logical tests, or introduce a new test method for clarity

Reply:
I wanted to check that the metadata api is called after expiring the cache. This code path is probably tested elsewhere but I wanted to be explicit about the expected behavior when an error causes the cache to be expired